lib/treewalk/splitter.js

```diff
@@ -71,12 +77,17 @@ export class TreewalkCarSplitter {
     channel.writer.put(block)
 
     for (const [, cid] of block.links()) {
+      // @ts-ignore TS does not get type of async generator
       for await (const val of this._cars(cid, parents, channel)) {
         channel = val.channel
         yield val
       }
     }
 
+    if (!channel) {
+      throw new Error('no blocks in the car')
+    }
+
```

Comment:
This one is unnecessary - there is always a channel (L66)

```suggestion
```

Reply:
ts complaint yah, I added the suggested error message